### PR TITLE
DataFrame API: Add ExperimentFromDev to namespace tb.data.experimental.*

### PR DESCRIPTION
**Pull request by caisq**
* Motivation for features / changes
  * Put the experimental `ExperimentFromDev` API (accessing tensorboard.dev
    scalars as `pandas.DataFrame`s) under a public import path:
    `tensorboard.data.experimental.ExperimentFromDev`.
* Technical description of changes
  * Add the necessary `__init__.py` and the associated BUILD targets.
  * In build_pip_package.sh, add the line `pip install -r requirements.txt`,
     because we know have lines in the files imported from the
     `__init__.py` files that import third-party dependencies such as grpc and
     numpy.

### CODE REVIEW

File: tensorboard/pip_package/build_pip_package.sh [hunk 1]
```diff
@@ -110,6 +110,7 @@ build() (
   # env markers are handled (https://github.com/pypa/setuptools/pull/1081)
   export PYTHONWARNINGS=ignore:DEPRECATION  # suppress Python 2.7 deprecation spam
   pip install -qU wheel 'setuptools>=36.2.0'
+  pip install -r requirements.txt
```
Comment (wchargin) [inline tensorboard/pip_package/build_pip_package.sh:113]: We shouldn’t need `grpc`, etc. just to build the Pip package, just as we
currently don’t need to install requirements even though TensorBoard
code depends on lots of other non-vendored packages. (And of course the
rest of TensorBoard already uses numpy.) Could you please define `data`
as a lazy module, like all the other nontrivial public exports, to avoid
this?

File: tensorboard/data/experimental/BUILD [hunk 2]
```diff
@@ -10,13 +10,13 @@ exports_files(["LICENSE"])
 py_library(
     name = "base_experiment",
     srcs = ["base_experiment.py"],
-    srcs_version = "PY3",
+    srcs_version = "PY2AND3",
```
Comment (wchargin) [inline tensorboard/data/experimental/BUILD:13]: Why this change? It doesn’t look like there are any Google-internal
reverse dependencies of `//tensorboard:lib` that are still Python 2
binaries. This change is okay with me if it’s really necessary for some
reason, but in general we’ve been preferring to write new code as py3.

File: tensorboard/pip_package/build_pip_package.sh (hunk 1 above)
Comment (wchargin) [reply tensorboard/pip_package/build_pip_package.sh:113]: The `pip install -r requirements.txt` appears to still be here.

File: tensorboard/BUILD [hunk 3]
```diff
@@ -64,6 +64,7 @@ py_library(
         ":errors",
         ":lazy",
         ":version",
+        "//tensorboard/data:lib_init_only",
```
[the remark this answers is not in the document]
Comment (caisq) [reply tensorboard/BUILD:67]: Done.

File: tensorboard/data/experimental/BUILD (hunk 2 above)
Comment (caisq) [reply tensorboard/data/experimental/BUILD:13]: Yeah, see the comment I sent yesterday. This change is necessary because some internal build targets are PY2. Specifically the notebook-related builds. I discovered breakages in those builds during a test sync of this PR, which prompted this change.

File: tensorboard/pip_package/build_pip_package.sh (hunk 1 above)
Comment (caisq) [reply tensorboard/pip_package/build_pip_package.sh:113]: Sorry I missed that in the previous revision. Removed now.

**Comment by caisq** [comment]
Gentle ping.

File: tensorboard/data/experimental/BUILD (hunk 2 above)
Comment (caisq) [reply tensorboard/data/experimental/BUILD]: After some further testing, I confirmed that the `PY2AND3` part isn't necessary. I have pushed a new commit to change them all to `PY3` in this file. PTAL @wchargin
